File: woodroid/src/main/java/me/gilo/woodroid/data/ProductReviewAPI.java
package me.gilo.woodroid.data;


import me.gilo.woodroid.data.callbacks.ReviewsData;
import me.gilo.woodroid.models.Coupon;
import me.gilo.woodroid.models.ProductReview;
import retrofit2.Call;
import retrofit2.http.*;

import java.util.List;

public interface ProductReviewAPI {

    @Headers("Content-Type: application/json")
    @POST("products/reviews")
    Call<ProductReview> create(@Body ProductReview body);

    @GET("products/reviews/{id}")
    Call<ProductReview> view(@Path("id") int id);

    @GET("products/reviews")
    Call<List<ProductReview>> list();

    @Headers("Content-Type: application/json")
    @PUT("products/reviews/{id}")
    Call<ProductReview> update(@Path("id") int id, @Body ProductReview body);

    @DELETE("products/reviews/{id}")
    Call<ProductReview> delete(@Path("id") int id);

    @DELETE("products/reviews/{id}")
    Call<ProductReview> delete(@Path("id") int id, @Query("force") boolean force);

    @POST("products/reviews/batch")
    Call<String> batch(@Body ProductReview body);

}

File: woodroid/src/main/java/me/gilo/woodroid/data/api/ProductReviewAPI.kt
package me.gilo.woodroid.data.api


import me.gilo.woodroid.models.ProductReview
import retrofit2.Call
import retrofit2.http.*

interface ProductReviewAPI {

    @Headers("Content-Type: application/json")
    @POST("products/reviews")
    fun create(@Body body: ProductReview): Call<ProductReview>

    @GET("products/reviews/{id}")
    fun view(@Path("id") id: Int): Call<ProductReview>

    @GET("products/reviews")
    fun list(): Call<List<ProductReview>>

    @Headers("Content-Type: application/json")
    @PUT("products/reviews/{id}")
    fun update(@Path("id") id: Int, @Body body: ProductReview): Call<ProductReview>

    @DELETE("products/reviews/{id}")
    fun delete(@Path("id") id: Int): Call<ProductReview>

    @DELETE("products/reviews/{id}")
    fun delete(@Path("id") id: Int, @Query("force") force: Boolean): Call<ProductReview>

    @POST("products/reviews/batch")
    fun batch(@Body body: ProductReview): Call<String>

    @GET("products/reviews")
    fun filter(@QueryMap filter: Map<String, String>): Call<List<ProductReview>>

}